lib/state-store.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * StateStore stores state and allows an observer to subscribe to state updates
+ */
+export default class StateStore {
+  constructor() {
+    this._listeners = [];
+    this._state = {};
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  update(props) {
+    Object.assign(this._state, props);
+    this._listeners.forEach(listener => listener());
+  }
+
+  subscribeUpdates(listener) {
+    this._listeners.push(listener);
+    return () => {
+      this._listeners = this._listeners.filter(l => l !== listener);
+    };
```

Comment:
Personally think a matching `unsubscribeUpdates` method would be a lot more clear than returning this unsubscribe here and storing it in ControlledComponent. That method could take `listener` as an arg, so you'd store a reference to that function in order to unsubscribe later